lib/mpl_toolkits/mplot3d/art3d.py

```diff
@@ -733,7 +733,13 @@ def set_zsort(self, zsort):
             The function applied on the z-coordinates of the vertices in the
             viewer's coordinate system, to determine the z-order.
         """
-        self._zsortfunc = self._zsort_functions[zsort]
+        def nansafe(func):
+            def f(x):
+                value = func(x)
+                return np.inf if np.isnan(value) else value
```

Comment:
Is this actually correct behavior? Would it be better to remove the NaNs before computing the mean?

Reply:
Hmmm yes I think we need both. I was just looking at plots where polys containing NaNs are completely masked, but in cases where you have just one masked corner the rendering is still wrong. One way to solve this would be to replace the `_zsort_functions` with their `nan` equivalents and then work around the `RuntimeWarning` raised if all elements of the tested array are `NaN`. 

So https://github.com/matplotlib/matplotlib/blob/498bcdda09058db2fd1aca9cc2437e5f2fa26055/lib/mpl_toolkits/mplot3d/art3d.py#L720-L724

would be replaced with 
```python
 _zsort_functions = { 
     'average': np.nanmean,   # mean and average the same with unweighted data?
     'min': np.nanmin, 
     'max': np.nanmax, 
 } 
```
and then we'd rewrite the `nansafe` wrapper (or the method like you suggest) to be 
```python
def nansafe(func):
    def f(x):
        return np.inf if np.isnan(x).all() else func(x)
    return f
```
Do you think that would be a solution? Do you see any problem with replacing the functions inside ` _zsort_functions`?

Reply:
I haven't had time to look into the details here. But keep in mind that the sorting is done everytime you rotate the view, which we want to be a real-time response. No question that filtering nans is necessary, but if you have multiple ways of doing that, please evaluate the performance.